### ISSUE TITLE
Pubsub: Client config is not propagated to the publisher

### ISSUE THREAD

**Issue by apoloval** [googleapis/google-cloud-ruby#2406]
In Pubsub initialization, it's possible to provide a client config that will override some of the settings at the GAX level. This is how the client config is propagated across the different layers:
* In `Google::Cloud::Pubsub.new` ([code](https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/master/google-cloud-pubsub/lib/google/cloud/pubsub.rb#L563-L565)), the `client_config` param is received. There, it's used to invoke the constructor of `Google::Cloud::Pubsub::Service`. 
* In `Google::Cloud::Pubsub::Service.initialize` ([code](https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/master/google-cloud-pubsub/lib/google/cloud/pubsub/service.rb#L40)), the `client_config` param is stored in a object attribute. 
* In `Google::Cloud::Pubsub::Service.subscriber`, the service initializes and memoizes the subscriber. The client config is used in the initialization (that's correct):
```ruby
        def subscriber
          return mocked_subscriber if mocked_subscriber
          @subscriber ||= begin
            V1::SubscriberClient.new(
              credentials: channel,
              timeout: timeout,
              client_config: client_config,
              lib_name: "gccl",
              lib_version: Google::Cloud::Pubsub::VERSION
            )
          end
        end
```
* Nevertheless, in `Google::Cloud::Pubsub::Service.publisher`, the `client_config` param is not specified:
```ruby
        def publisher
          return mocked_publisher if mocked_publisher
          @publisher ||= begin
            V1::PublisherClient.new(
              credentials: channel,
              timeout: timeout,
              lib_name: "gccl",
              lib_version: Google::Cloud::Pubsub::VERSION
            )
          end
        end
```

This is causing the **publisher to not receive the client config**. As result, it's not possible to configure client settings for publishing (in particular, backoff/retry settings).

**Comment by apoloval** [comment]
It seems @quartzmo forgot to include the param in commit 580b8feb16e980f50327cba20d88a325712e689e.

**Comment by blowmage** [comment]
Might be a good idea to double-check the other gems to make sure this isn’t happening anywhere else.

**Comment by quartzmo** [comment]
I checked (including generated code), but I didn't find any usages missing `client_config: client_config` other than the Pub/Sub emulator case [mentioned here](https://github.com/GoogleCloudPlatform/google-cloud-ruby/pull/2407#issuecomment-418773945).

**Comment by blowmage** [comment]
This was released in [google-cloud-pubsub 0.32.1](https://github.com/googleapis/google-cloud-ruby/releases/tag/google-cloud-pubsub/v0.32.1).
